packages/cadl-ranch-specs/http/azure/core/traits/mockapi.ts

```diff
@@ -17,8 +17,15 @@ Scenarios.Azure_Core_Traits_smokeTest = passOnSuccess(
       throw new ValidationError("Expected path param id=1", "1", req.params.id);
     }
     req.expect.containsHeader("foo", "123");
-    req.expect.containsHeader("if-match", '"valid"');
-    req.expect.containsHeader("if-none-match", '"invalid"');
+    const if_none_match = req.headers["if-none-match"];
+    const if_match = req.headers["if-match"];
```

Comment:
Does `req.headers` auto convert header name to lowercase?

Reply:
I guess case insensitive since `req.expect.containsHeader` doesn't check case either:
https://github.com/Azure/cadl-ranch/blob/b63ed94fd07a5070a0589d941b5cb6ec84e8763a/packages/cadl-ranch-api/src/request-validations.ts#L85-L90